### ISSUE TITLE
Issue with BlobRequestOptions.setRequireEncryption

### ISSUE THREAD

**Issue by jackmorris2** [Azure/azure-storage-java#143]
When I upload a blob without encryption and then I try download part of it with `downloadRangeToByteArray` with requiereEncryption set to true the code is stuck (it should throw).

Here is a sample of code to reproduce the issue:

```
CloudBlobClient client = BlobUtil.getCloudBlobClient("UseDevelopmentStorage=true");
CloudBlobContainer container = client.getContainerReference("container");
container.createIfNotExists();

CloudBlockBlob blob = container.getBlockBlobReference("blob");

byte[] msg = "my message".getBytes();
// Upload data without encryption
blob.uploadFromByteArray(msg, 0, msg.length);

// Create options with encryption policy
BlobRequestOptions options = new BlobRequestOptions();
options.setEncryptionPolicy(new BlobEncryptionPolicy(new RsaKey("myKey", 1024), null));
options.setRequireEncryption(true);

// This works
try {
	blob.downloadText(Charset.defaultCharset().name(), null, options, null);
} catch (Exception e) {
	System.out.println("downloadText failed as expected");
}

byte[] buffer = new byte[msg.length];
System.out.println("before downloadRangeToByteArray");
// The code is stuck here
blob.downloadRangeToByteArray(0, (long) buffer.length, buffer, 0, null, options, null);
System.out.println("downloadRangeToByteArray completed");

System.out.println(new String(buffer));
```

**Comment by jofriedm-msft** [comment]
Thanks for the repro code @jackmorris2 , we will take a look.

**Comment by jofriedm-msft** [comment]
@jackmorris2 , thanks for raising this issue! We have found the issue and it will be fixed in our next release 😎

**Comment by jofriedm-msft** [comment]
This should be fixed in the 5.1.0 release
